### ISSUE TITLE
[docs/README.RaspberryPi.md] Building tools results in "Error 2"

### ISSUE THREAD

**Issue by mdPlusPlus** [xbmc/xbmc#18078]
---
name: [docs/README.RaspberryPi.md] Building tools results in "Error 2"
about: Following the instructions to build Kodi for RPi3 does not result in success.

---
### Describe the bug

Following the instructions to build Kodi for Raspberry Pi 3 under Ubuntu results in `Error 2` during [step 4](https://github.com/xbmc/xbmc/blob/master/docs/README.RaspberryPi.md#4-build-tools-and-dependencies).

## Expected Behavior

Building the dependencies and tools should result in success without errors.

## Actual Behavior

`make -j$(getconf _NPROCESSORS_ONLN)` returns `make: *** [Makefile:11: target/.installed-raspberry-pi2-release] Error 2` without any other obvious errors in the build log.

Tested with `master` and `Leia` branch of Kodi, and `master` and `stable` branch of raspberrypi/firmware.

## Possible Fix

Unfortunately I am not qualified to propose a fix.

### To Reproduce

    cd /mnt/8tb/
    mkdir kodi-rpios
    cd kodi-rpios/
    sudo apt install autoconf bison build-essential curl default-jdk gawk git gperf libcurl4-openssl-dev zlib1g-dev
    git clone -b Leia https://github.com/xbmc/xbmc kodi
    git clone https://github.com/raspberrypi/tools --depth=1
    git clone -b stable https://github.com/raspberrypi/firmware --depth=1
    mkdir kodi-rpi
    cd kodi/tools/depends/
    ./bootstrap
    ./configure --host=arm-linux-gnueabihf --prefix=/mnt/8tb/kodi-rpios/kodi-rpi --with-toolchain=/mnt/8tb/kodi-rpios/tools/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf --with-firmware=/mnt/8tb/kodi-rpios/firmware --with-platform=raspberry-pi2 --disable-debug
    make -j$(getconf _NPROCESSORS_ONLN)

### Debuglog
The debuglog can be found here:

Build log (in German unfortunately): https://gofile.io/d/XBxcTi

### Screenshots 
Here are some links or screenshots to help explain the problem:

N/A

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

N/A

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Ubuntu 20.04
 - Kodi version: Leia

**Comment by mdPlusPlus** [comment]
Applied template.

**Comment by mdPlusPlus** [comment]
After checking the log again, I stumbled up on this:

    make -C raspberry-pi2-release/build install
    make[3]: Verzeichnis „/mnt/8tb/kodi-rpios/kodi/tools/depends/target/libfmt/raspberry-pi2-release/build“ wird betreten
    strerror-sym.c: In function 'gpg_strerror_sym':
    strerror-sym.c:45:7: warning: implicit declaration of function 'errnos_msgidxof' [-Wimplicit-function-declaration]
           idx = errnos_msgidxof (code);
           ^
    strerror-sym.c:47:9: error: 'errnos_msgstr' undeclared (first use in this function)
      return errnos_msgstr + errnos_msgidx[idx];
             ^
    strerror-sym.c:47:9: note: each undeclared identifier is reported only once for each function it appears in
    strerror-sym.c:47:25: error: 'errnos_msgidx' undeclared (first use in this function)
      return errnos_msgstr + errnos_msgidx[idx];
                             ^
    make[4]: Verzeichnis „/mnt/8tb/kodi-rpios/kodi/tools/depends/target/pcre/raspberry-pi2-release“ wird verlassen
    make[6]: *** [Makefile:946: gpg_error-strerror-sym.o] Fehler 1

**Comment by tomvdbroek** [comment]
The problem lies in the environment. The makefile generated by configure uses a variable name for a gawk internal keyword. Namely: "namespace".

Proposed solution: redefine the variable to a new name.

**Comment by mdPlusPlus** [comment]
Any indicator for when this will be merged into the `Leia` branch or should I compile `master` instead?

**Comment by wsnipex** [comment]
the build issue on Leia should by fixed by https://github.com/xbmc/xbmc/pull/18198
